Elmish.XamarinForms/ViewConverters.fs

```diff
@@ -135,8 +135,27 @@ module Converters =
 
     let identical (x: 'T) (y:'T) = System.Object.ReferenceEquals(x, y)
 
-    let canReuseChild (prevChild:ViewElement) (newChild:ViewElement) = 
-        (prevChild.TargetType = newChild.TargetType)
+    // NavigationPage can be reused only if the already existing pages don't change type (added/removed pages don't prevent reuse)
+    // E.g. If the first page switch from ContentPage to TabbedPage, the NavigationPage can't be reused; otherwise it can.
+    let canReuseNavigationPage (prevChild:ViewElement) (newChild:ViewElement) =
+        let prevPages = prevChild.TryGetAttribute<ViewElement[]>("Pages")
+        let newPages = newChild.TryGetAttribute<ViewElement[]>("Pages")
+
+        match prevPages, newPages with
+        | ValueSome prevPages, ValueSome newPages ->
+            let prevSeq = Seq.ofArray prevPages
+            let newSeq = Seq.ofArray newPages
+            Seq.forall2 (fun (prev: ViewElement) (curr: ViewElement) -> prev.TargetType = curr.TargetType) prevSeq newSeq
+        | _, _ -> true
+
+    let canReuseChild (prevChild:ViewElement) (newChild:ViewElement) =
+        if prevChild.TargetType = newChild.TargetType then
+            if newChild.TargetType = typeof<NavigationPage> then
```

Comment:
I think this should use `IsAssignableFrom` (a subtyping test) rather than strict type equality, just in case someone defines a subclass of `NavigationPage`

Reply:
Ok